service/app.js

```diff
@@ -19,407 +30,40 @@ try {
-      res.sendStatus(404)
-    }
-  } catch (err) {
-    console.log(`error in creating card ${err}`)
-    res.sendStatus(502)
-  }
-})
-
-
-// update Deck
-app.put("/updateDeck/:id", async (req, res) => {
-  try {
-    const deck = await Deck.findById(req.params.id)
-    if(deck) {
-      deck.name = req.body.name
-      if(req.body.cards) {
-        deck.cards = req.body.cards
-        deck.cards = req.body.size
-      }
-      if(req.body.userId) {
-        deck.userId = req.body.userId
-      }
-    } else {
-      return res.status(404).json({
-        error: "deck does not exists"
-      })
-    }
-    const result = await deck.save()
-    return res.status(201).json({
-      msg: result
-  })
-  } catch (error) {
-    console.log(error)
-  }
-})
-
-// update user
-app.put("/updateUser/:id", async (req, res) => {
-  try {
-    const user = await User.findById(req.params.id)
-    if(user) {
-      user.name = req.body.name
-    } else {
-      return res.status(404).json({
-        error: "User does not exists"
-      })
-    }
-    const result = await user.save()
-    return res.status(201).json({
-      msg: result
-    })
-  } catch (error) {
-    console.log(error)
-  }
-})
-
-// delete card
-app.delete('/deletecard/:deckId/:cardId', async (req, res) => {
-  if (!req.params.deckId) {
-    res.status(400).send('Deck ID is required')
-  }
-
-  try {
-    const deck = await Deck.findById(req.params.deckId)
-    if (deck) {
-      deck.cards.forEach(card => {
-        if(card._id == req.params.cardId) {
-          card.remove()
-        }
-      });
-      deck.size -= 1
-      await deck.save()
-      res.sendStatus(204)
-    } else {
-      res.sendStatus(404)
-    }
-  } catch (err) {
-    console.log(`error in creating card ${err}`)
-    res.sendStatus(502)
-  }
-})
-
-// delete a Deck and all asscociated cards
-app.delete("/deleteDeck/:id", async (req, res) => {
-  try {
-    const deck = await Deck.findById(req.params.id)
-    if(deck) {
-      await deck.remove()
-    } else {
-      return res.status(404).json({
-        error: "deck does not exists"
-      })
-    }
-    return res.status(200).json({
-      msg: "deck deleted"
-    })
-  } catch (error) {
-    console.log(error)
-  }
-})
-
-// delete user
-app.delete("/user/:id", async (req, res) => {
-  try {
-    const user = await User.findById(req.params.id)
-    if(user) {
-      await user.remove()
-    } else {
-      return res.status(404).json({
-        error: "user does not exists"
-      })
-    }
-    return res.status(201).json({
-      msg: "user deleted"
-    })
-  } catch (error) {
-    return res.status(404).json({
-      error: "user does not exists"
-    })
-  }
-})
+// Deck Routes
+app.get('/decks', getDecks)
+app.get('/decks/:id', deckById)
+app.post('/decks', body('name').not().isEmpty(), createDeck)
+app.put(
+  '/decks/:id',
+  body('name').not().isEmpty(),
+  updateDeck
+)
+app.delete('/decks/:id', deleteDeck)
+
+app.post(
+  '/decks/:id/cards',
```

Comment:
This route accepts and saves to the database without completely validating the request.